src/main/java/service/activity/SupplyHandler.java

```diff
@@ -0,0 +1,12 @@
+package service.activity;
+
+import java.util.Map;
+import model.FruitRecord;
+
+public class SupplyHandler implements ActivityHandler {
```

Comment:
Your supply and return handlers are the same, let's create one generic handler and use for both  return and supply operations